src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLPaints.m

```diff
@@ -948,6 +967,16 @@ - (void)setPipelineState:(id <MTLRenderCommandEncoder>)encoder
                                                                              renderOptions:renderOptions
                                                                              stencilNeeded:[mtlc.clip isShape]];
         [encoder setRenderPipelineState:pipelineState];
+    } else {
+      // Fallback to valid pipeline state
```

Comment:
We should not call it a valid pipeline state. It is more of standard backup pipeline state with color 0.

We are hitting default MTLPaint without any texture(which seems to be the main root cause for this issue- state management problem) and we end up not setting any pipeline state. Please reword this comment.

Reply:
Also is it possible for us to just return at any earlier stage before we hit setRenderPipelineState() without valid texture or color?